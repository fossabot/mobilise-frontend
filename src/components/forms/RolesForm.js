import React from 'react';
import { Form, Row, Col } from 'react-bootstrap';
import { Typeahead, Token } from 'react-bootstrap-typeahead';

class RolesForm extends React.Component {
  _renderToken = (option, props, index) => (
    <Token key={index} style={{}} onRemove={props.onRemove}>
      <Row>
        <Col
          md="auto"
          style={{
            padding: '0 0.2rem 0 1rem',
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center'
          }}
        >
          {option.label ? option.label : option}
        </Col>
        <Col
          style={{
            padding: '0 1rem 0 0.2rem',
            width: '3.2rem'
          }}
        >
          <Form.Control
            type="number"
            style={{ height: '1.4rem', textAlign: 'center' }}
          />
        </Col>
      </Row>
    </Token>
  );

  render() {
    const { roleOptions, handleChange } = this.props;
    return (
      <Form.Group controlId="rolesForm">
        <Form.Label>Roles</Form.Label>
        <Typeahead // TODO make async & SORT OUT CSS so letters like g dont get cut off.
          renderToken={this._renderToken}
          id="roles"
          placeholder="Add available roles for shift"
          newSelectionPrefix="Add new role:  "
          options={roleOptions.map(r => r.name)}
          allowNew
          multiple
          onChange={handleChange}
        />
      </Form.Group>
    );
  }
}

export default RolesForm;
