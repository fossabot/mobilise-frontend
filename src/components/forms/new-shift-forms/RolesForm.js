import React from 'react';
import { Form } from 'react-bootstrap';
import { Typeahead, Token } from 'react-bootstrap-typeahead';
// import NumericInput from 'react-numeric-input';

export default class RolesForm extends React.Component {
  submit = () => console.log('Roles inputted'); // TODO create submit function

  _renderToken = (option, props, index) => {
    return (
      <Token key={index} onRemove={props.onRemove}>
        {`${option}\t`}
        {/* CREATES WARNING WHEN NUMBER ALTERED <NumericInput min={0} max={100} size={1} /> */}
      </Token>
    );
  };

  render() {
    return (
      <Form id="new-shift-form-part-3" onSubmit={this.submit}>
        <Form.Group controlId="locationForm">
          <Form.Label>Location</Form.Label>
          <Form.Control
            required
            type="location"
            placeholder="e.g. Imperial College London"
          />
          {/* TODO: use google maps API & asyncTypeAhead */}
        </Form.Group>
        <Form.Group controlId="rolesForm">
          <Form.Label>Roles</Form.Label>
          <Typeahead // TODO make async
            renderToken={this._renderToken}
            id="Roles"
            bodyContainer
            clearButton
            placeholder="Add available roles for shift"
            options={["driver's mate", 'warehouse assistant']}
            allowNew
            multiple
          />
        </Form.Group>
      </Form>
    );
  }
}
