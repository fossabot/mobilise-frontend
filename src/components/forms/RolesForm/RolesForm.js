import React from 'react';
import { Form, Row, Col } from 'react-bootstrap';
import { Typeahead, Token } from 'react-bootstrap-typeahead';
import './RolesForm.css';

class RolesForm extends React.Component {
  _renderToken = (option, props, index, handleRoleNumber, roles) => {
    if (roles.length === 0) {
      return null;
    }
    const roleName = option.customOption ? option.label : option;
    const roleToChange = roles.find(r => r.roleName === roleName);
    if (!roleToChange) {
      return null;
    }
    return (
      <Token key={index} onRemove={props.onRemove}>
        <Row>
          <Col md="auto" className="role-name">
            {roleName}
          </Col>
          <Col className="role-num-col">
            <Form.Control
              name={roleName}
              type="number"
              className="role-num-form"
              min={1}
              onChange={handleRoleNumber}
              value={roleToChange.number}
            />
          </Col>
        </Row>
      </Token>
    );
  };

  render() {
    const { roles, roleOptions, handleChange, handleRoleNumber } = this.props;
    return (
      <Form.Group controlId="rolesForm">
        <Form.Label>Roles</Form.Label>
        <Typeahead // TODO make async & SORT OUT CSS so letters like g dont get cut off.
          id="roles"
          placeholder="Add available roles for shift"
          newSelectionPrefix="Add new role:  "
          options={roleOptions.map(r => r.name)}
          allowNew
          multiple
          onChange={handleChange}
          selected={roles.map(r => r.name)}
          selectHintOnEnter
          renderToken={(option, props, index) =>
            this._renderToken(option, props, index, handleRoleNumber, roles)
          }
        />
      </Form.Group>
    );
  }
}

export default RolesForm;
