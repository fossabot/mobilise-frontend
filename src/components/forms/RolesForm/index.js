import RolesForm from './RolesForm';

export default RolesForm;
