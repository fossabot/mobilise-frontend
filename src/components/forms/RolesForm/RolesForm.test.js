import React from 'react';
import { shallow } from 'enzyme';
import RolesForm from './RolesForm';

const testRoleProps = {
  roles: [],
  roleOptions: [],
  handleChange: () => {},
  handleRoleNumber: () => {}
};

let wrapper;

beforeEach(() => {
  wrapper = shallow(<RolesForm {...testRoleProps} />);
});

describe('Roles Form Component', () => {
  it('renders form correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain a typeahead', () => {
    expect(wrapper.find('TypeaheadComponent')).toHaveLength(1);
  });

  it('contains function to render tokens', () => {
    expect(wrapper.find('_renderToken')).toBeDefined();
  });
});
