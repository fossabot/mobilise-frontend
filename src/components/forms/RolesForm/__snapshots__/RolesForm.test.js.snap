// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roles Form Component renders form correctly 1`] = `ShallowWrapper {}`;
